### DISCUSSION TITLE
Edit inputs from an ONNX model

### DISCUSSION THREAD

**Discussion by IzanCatalan** [microsoft/onnxruntime#12326] [Other Q&A]
Hi all!, Does anyone know how to edit inputs from an onnx model in python? 

I would like to change for exameple one element of an input tensor (like weights in a convolution layer) in one node of the graph. I tried to do it with " model = onnx.load ()" and after with "inputs = model.graph.initializers" but when I get de inputs, I can read them but not write them, they're read only.

is there any way to write them, using another tool for instance? Thanks.

**Comment by tianleiwu** [comment]
You can write them directly (I mean change data member like data_type, raw_data etc. of initializer).

For example, we have a tool to convert model from float32 to float16. That need change initializers of float32. Related code:
https://github.com/microsoft/onnxruntime/blob/bdd6b00c9a6a5cdd2925762a9058e5c46f4a787f/onnxruntime/python/tools/transformers/float16.py#L68-L83
